7.0 release notes preview (#10)
ran-huang/pingcap-docs-website-scaffold: markdown-pages/en/tidb/master/releases/release-7.0.0.md

    Before integrating TiDB Lightning, the `LOAD DATA` statement could only import data files from the client side. If you wanted to import data from cloud storage, you had to rely on TiDB Lightning. However, deploying TiDB Lightning separately would incur additional deployment and management costs. By integrating the TiDB Lightning logical import capability (TiDB Backend) into the `LOAD DATA` statement, you can not only eliminate the deployment and management costs of TiDB Lightning, but also greatly extend the capabilities of the `LOAD DATA` statement with the TiDB Lightning features. Some examples of the extended functionality are as follows:

    - Supports importing data from Amazon S3 and Google Cloud Storage to TiDB. Supports importing multiple source files to TiDB in one go with wildcards.
    - Support using `DEFINED NULL BY` to define null.
    - Support source files in CSV, TSV, Parquet, and SQL (mydumper/dumpling) formats.
    - Support setting tasks as `Detached` to let tasks run in the background.
    - Support task management. You can query task status and progress details by `SHOW LOAD DATA jobid`, which is convenient for management and maintenance.

>     For more information, see [documentation](/sql-statements/sql-statement-load-data.md).

Reviewer: ```suggestion
  For more information, see [documentation](/sql-statements/sql-statement-load-data.md).
```

---
title: TiDB 7.0.0 Release Notes
---

# TiDB 7.0.0  Release Notes

Release date: xx, 2023

TiDB version: 7.0.0-DMR

Quick access: Quick start | Installation package

> In v7.0.0-DMR, the key new features and improvements are as follows:

Reviewer 2: ```suggestion
In 7.0.0-DMR, the key new features and improvements are as follows:
```

<table>
<thead>
  <tr>
    <th>Category</th>

>     <th>Feature/Enhancement</th>

Author: ```suggestion
    <th>Feature</th>
```

    <th>Description</th>
  </tr>
</thead>
<tbody>
  <tr>
    <td rowspan="2">Scalability and Performance<br/></td>
    <td>Session level SQL plan cache (experimental)</td>

>     <td>Using plan cache can reduce the time for compiling and optimizing SQL queries, which reduces query latency. For each query executed, if a plan doesn't exist, a new one will be stored and re-used next time.</td>

Reviewer 3: Automatically re-using plan cache can reduce the time for compiling and optimizing same SQL query patterns, which can reduce query latency without manually set prepare statement in advance.

Reviewer 3: change feature name to 'Session level non-prepared SQL plan cache'

>     <td>Using plan cache can reduce the time for compiling and optimizing SQL queries, which reduces query latency. For each query executed, if a plan doesn't exist, a new one will be stored and re-used next time.</td>

Author: ```suggestion
    <td>Automatically re-using plan cache can reduce the time for compiling and optimizing the same SQL query patterns, which can reduce query latency without manually setting prepare statements in advance.</td>
```

>     <td>Automatically re-using plan cache can reduce the time for compiling and optimizing the same SQL query patterns, which can reduce query latency without manually setting prepare statements in advance.</td>

Reviewer 2: ```suggestion
    <td>Support automatically reusing plan cache at the session level to reduce compilation and shorten the query time for the same SQL patterns without manually setting prepare statements in advance.</td>
```

  </tr>
  <tr>

>     <td>TiFlash supports storage-computing separation and S3 shared storage (experimental)</td>

Reviewer 2: ```suggestion
    <td>TiFlash supports the disaggregated storage and compute architecture and S3 shared storage (experimental)</td>
```

>     <td>TiFlash introduces a new architecture:

Reviewer 3: TiFlash introduces a cloud-native architecture as an option:

      <ul>

>         <li>Disaggregates TiFlash's compute and storage which is a milestone for elastic HTAP resource utilization.</li>

Reviewer 2: ```suggestion
        <li>Disaggregates TiFlash's compute and storage, which is a milestone for elastic HTAP resource utilization.</li>
```

>         <li>Introduces S3 based storage engine which can provide shared storage at lower cost.</li>

Reviewer 2: ```suggestion
        <li>Introduces S3-based storage engine, which can provide shared storage at a lower cost.</li>
```

      </ul>
    </td>
  </tr>
  <tr>
    <td rowspan="2">Reliability and Availability<br/></td>

>     <td>Resource control enhancement (GA) </td>

Author: ```suggestion
    <td>Resource control enhancement (Experimental) </td>
```

>     <td>Users can now use resource groups to allocate and isolate resources for different applications or workloads in one cluster. This supports different resource binding modes (user, session, query) and user-defined priority. Quotas can be "burstable".</td>

Reviewer 3: Users now can use resource groups to allocate and isolate resources for different applications or workloads in one cluster. In this version, TiDB added supports for different resource binding modes (user, session, query) and user-defined priority, users can also use command to  do resource calibration(estimation for whole resource amount).

>     <td>Users now can use resource groups to allocate and isolate resources for various applications or workloads within one cluster. In this release, TiDB adds support for different resource binding modes (user, session, query) and user-defined priorities. Additionally, users can also use commands to perform resource calibration (estimation for the whole resource amount).</td>

Reviewer 2: ```suggestion
    <td>Support using resource groups to allocate and isolate resources for various applications or workloads within one cluster. In this release, TiDB adds support for different resource binding modes (user, session, and query level) and user-defined priorities. Additionally, you can also use commands to perform resource calibration (estimation for the whole resource amount).</td>
```

  </tr>
  <tr>

>     <td>TiFlash support spill to disk</td>

Reviewer 2: ```suggestion
    <td>TiFlash supports spill to disk</td>
```

>     <td>TiFlash supports intermediate result spill to disk to mitigate OOMs in data-intensive operations like aggregations, sorts and hash joins.</td>

Reviewer 2: ```suggestion
    <td>TiFlash supports intermediate result spill to disk to mitigate OOMs in data-intensive operations such as aggregations, sorts, and hash joins.</td>
```

  </tr>
  <tr>
    <td rowspan="2">SQL</td>
    <td>Row-level TTL (GA)</td>

>     <td>Manage database size and improve performance by automatically expiring data of a certain age.</td>

Reviewer 2: ```suggestion
    <td>Support managing database size and improve performance by automatically expiring data of a certain age.</td>
```

Reviewer 2: 如果不加 Support，感觉是自动就会过期，用户不需要手动设置 TTL

  </tr>
  <tr>
    <td>Reorganize <code>LIST</code>/<code>RANGE</code> partition</td>

>     <td><code>REORGANIZE PARTITION</code> clause can be used for merging adjacent partitions or splitting one to many, which  provides users with better usability of partitioned tables.</td>

Reviewer 2: ```suggestion
    <td>The <code>REORGANIZE PARTITION</code> statement can be used for merging adjacent partitions or splitting one partition into many, which provides better usability of partitioned tables.</td>
```

  </tr>
  <tr>
    <td rowspan="2">DB Operations and Observability<br/></td>

>     <td>TiDB integrates Lightning by “load data” statement(experimental)</td>

Reviewer 2: ```suggestion
    <td>The <code>LOAD DATA</code> statement integrates with TiDB Lightning (experimental) </td>
```

    <td>Load data sql statement becomes more powerful by integrating Lightning TiDB backend, such as supporting data import from S3/GCS, supporting task management and so on.<br/></td>
  </tr>
  <tr>
    <td>TiCDC supports object storage sink (GA)</td>

>     <td>TiCDC supports saving row change events to storage services, including Amazon S3, Azure Blob Storage, and NFS<br/></td>

Reviewer 3: TiCDC supports saving row change events to object storage services, including Amazon S3, Azure Blob Storage, and NFS.

>     <td>TiCDC supports saving row change events to object storage services, including Amazon S3, Azure Blob Storage, and NFS.<br/></td>

Reviewer 2: ```suggestion
    <td>TiCDC supports replicating row change events to object storage services, including Amazon S3, Azure Blob Storage, and NFS.<br/></td>
```
